### ISSUE TITLE
Request type in onResponse option

### ISSUE THREAD

**Issue by swarthy** [fastify/fastify-reply-from#224]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.24.1

### Plugin version

6.4.1

### Node.js version

16.x

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

20.04

### Description

Request time in index.d.ts (https://github.com/fastify/fastify-reply-from/blob/master/index.d.ts#L36) (FastifyRequest) doesn't match passed argument in code (https://github.com/fastify/fastify-reply-from/blob/8d8026f1050396f31f9ec09892562be682f786c3/index.js#L153) (Raw request)

### Steps to Reproduce

I'm trying to pass replyOptions for https://github.com/fastify/fastify-http-proxy
```
const options: proxy.FastifyHttpProxyOptions = {
        upstream,
        prefix,
        rewritePrefix,
        config: route,
        proxyPayloads: false,
        preHandler: (req, reply) => { Object.assign(req.raw, { startedAt: hrtime.bigint() }) },
        undici: {
          headersTimeout: this.headersTimeout,
        },
        replyOptions: {
          onResponse: (req, reply, res) => {
            // req type is FastifyRequest, but real type is IncommingMessage
            console.log(req.raw) // undefined
            console.log('time to first byte:', hrtime.bigint() - req.raw.startedAt)  // cant read startedAt of undefined
            reply.send(res);
          },
        },
      };
```

### Expected Behavior

First argument of `onResponse` should have correct type (`IncommingMessage` in [index.d.ts](https://github.com/fastify/fastify-reply-from/blob/master/index.d.ts#L36) or `onResponse(this.request, this, res.stream)` in [code](https://github.com/fastify/fastify-reply-from/blob/8d8026f1050396f31f9ec09892562be682f786c3/index.js#L153)

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by swarthy** [comment]
Yes. Should I make backward compatible changes (type in index.d.ts) or change `onResponse(this.request.raw, this, res.stream) to onResponse(this.request, this, res.stream)`?

**Comment by swarthy** [comment]
Fixed in #225
